src/analysis_file.py

```diff
@@ -0,0 +1,216 @@
+                    for plan in PipelineConfiguration.DEMOGS_CODING_PLANS 
+                    if plan.analysis_file_key is not None}, 
+                    Metadata(user, Metadata.get_call_location(), time.time())
+            )
+        
+        # Convert RQA binary codes to their string values
+        for td in data:
+            td.append_data(
+                {plan.binary_analysis_file_key:
+                plan.binary_code_scheme.get_code_with_id(td[plan.binary_coded_field]["CodeID"]).string_value
+                for plan in PipelineConfiguration.RQA_CODING_PLANS if plan.binary_code_scheme is not None},
+                Metadata(user, Metadata.get_call_location(), time.time())
+            )
+        
+        # Translate the RQA reason codes to matrix values
+        matrix_keys = []
+
+        for plan in PipelineConfiguration.FOLLOW_UP_CODING_PLANS + PipelineConfiguration.RQA_CODING_PLANS:
+            show_matrix_keys = list()
+            for code in plan.code_scheme.codes:
+                show_matrix_keys.append(f"{plan.analysis_file_key}{code.string_value}")
+        
+            AnalysisKeys.set_matrix_keys(
+                user, data, show_matrix_keys, plan.code_scheme, plan.coded_field, plan.analysis_file_key)
+            matrix_keys.extend(show_matrix_keys)
+             
+        binary_keys = [plan.binary_analysis_file_key
+                       for plan in PipelineConfiguration.RQA_CODING_PLANS
+                        if plan.binary_analysis_file_key is not None]
+        
+        equal_keys = ["uid"]
+        equal_keys.extend(demog_keys)
+        concat_keys = [plan.raw_field for plan in PipelineConfiguration.RQA_CODING_PLANS]
+        follow_up_concat_keys = [plan.raw_field for plan in PipelineConfiguration.FOLLOW_UP_CODING_PLANS]
+        concat_keys.extend(follow_up_concat_keys)
+        bool_keys = [
+            consent_withdrawn_key,
+            "radio_promo",
+            "radio_show",
+            "non_logical_time",
+            "radio_participation_s01e01",
+            "radio_participation_s01e02",
+            "radio_participation_s01e03",
+            "radio_participation_s01e04",
+            "radio_participation_s01e05",
+            "radio_participation_s01e06"
+        ]
+
+        # Export to CSV
+        export_keys = ["uid"]
+        export_keys.extend(bool_keys)
+        export_keys.extend(matrix_keys)
+        export_keys.extend(binary_keys)
+        export_keys.extend(concat_keys)
+        export_keys.extend(demog_keys)
+        
+        # Set consent withdrawn based on presence of data coded as "stop"
+        ConsentUtils.determine_consent_withdrawn(
+            user, data, PipelineConfiguration.DEMOGS_CODING_PLANS, consent_withdrawn_key)
+    
+        # Set consent withdrawn based on stop codes from radio question answers
+        for td in data:
+            for plan in PipelineConfiguration.RQA_CODING_PLANS:
```

Comment:
I think you need to apply this to the follow up surveys too

Reply:
similar loops merged thanks!